test/syscalls/linux/pty.cc

```diff
@@ -650,6 +650,25 @@ class PtyTest : public ::testing::Test {
     t.c_lflag |= ICANON;
     EXPECT_THAT(ioctl(replica_.get(), TCSETS, &t), SyscallSucceeds());
   }
+ 
+  // Writes master_input to the master file descriptor and verifies that
+  // the replica and the echo output match what is expected.
+  void TestCanonicalIO(const char* master_input, const char* expected_replica_output, const char* expected_echo_output) {
+    ASSERT_THAT(WriteFd(master_.get(), master_input, strlen(master_input)),
+                SyscallSucceedsWithValue(strlen(master_input)));
+
+    char buf[strlen(expected_replica_output)+1] = {};
```

Comment:
I believe variable-length arrays are nonstandard in C++. Maybe we can do something like:

```cpp
std::vector buf(strlen(expected_replica_output)+1, 0);
ASSERT_NO_ERRNO(WaitUntilReceived(replica_.get(), strlen(expected_replica_output)));
ExpectReadable(replica_, strlen(expected_replica_output), &buf[0]);
EXPECT_STREQ(&buf[0], expected_replica_output);
```

Same for `echo_buf`.

Reply:
or maybe just `std::string`?